tools/mascot/src/main/java/dk/alexandra/fresco/tools/mascot/field/FieldElementSerializer.java

```diff
@@ -0,0 +1,90 @@
+package dk.alexandra.fresco.tools.mascot.field;
+
+import dk.alexandra.fresco.framework.network.serializers.ByteSerializer;
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class FieldElementSerializer implements ByteSerializer<FieldElement> {
+
+  private final BigInteger modulus;
+  private final int modBitLength;
+  private final FieldElementUtils fieldElementUtils;
+
+  /**
+   * Creates new {@link FieldElementSerializer}.
+   * 
+   * @param modulus modulus of field elements
+   * @param modBitLength mod bit length
+   */
+  public FieldElementSerializer(BigInteger modulus, int modBitLength) {
+    this.modulus = modulus;
+    this.modBitLength = modBitLength;
+    this.fieldElementUtils = new FieldElementUtils(modulus, modBitLength);
+  }
+
+  /**
+   * Deserializes a single field element.
+   * 
+   * @param data serialized element
+   * @return deserialized field element
+   */
+  @Override
+  public FieldElement deserialize(byte[] data) {
+    return new FieldElement(data, modulus, modBitLength);
+  }
+
+  /**
+   * Serializes a single field element.
+   * 
+   * @param obj field element to serialize
+   * @return serialized field element
+   */
+  @Override
+  public byte[] serialize(FieldElement obj) {
+    return obj.toByteArray();
```

Comment:
We should check that the modulus of `obj` is consistent with this serializer. Otherwise, we may not be able to correctly deserialize the result.

Reply:
Done.